shell/utils/inactivity.ts

```diff
@@ -0,0 +1,183 @@
+          requiresFreshData = true;
+        }
+
+        // this is the POST to the UserActivity, which will reset the expireAt
+        if (requiresFreshData) {
+          userActivity = await updateUserActivityToken(store, sessionTokenName);
+        }
+
+        if (userActivity) {
+          const data = parseTTLData(userActivity);
+
+          console.debug(`UI inactivity modal (backend-based) will show after ${ data.showModalAfter || 0 / 60 }(m) and be shown for ${ data.courtesyTimer || 0 / 60 }(m)`); // eslint-disable-line no-console
+
+          return data;
+        }
+
+        console.error(`Could not find any user activity for token ${ sessionTokenName }`); // eslint-disable-line no-console
+
+        return defaultParsedTTLData;
+      } catch (e) {
+        console.error(`Could not update user activity for token ${ sessionTokenName }`, e); // eslint-disable-line no-console
+
+        return defaultParsedTTLData;
+      }
+    }
+  }
+
+  console.error('Cannot find a userActivity for this session OR this user does not have permissions to list the necessary resources'); // eslint-disable-line no-console
+
+  return defaultParsedTTLData;
+}
+
+export async function checkUserActivityData(store: any, sessionTokenName: string): Promise<userActivityResponse> {
+  try {
+    const updatedData = await store.dispatch('management/find', { type: EXT.USERACTIVITY, id: sessionTokenName });
+
+    // update stored data
+    storedUserActivityResponse = updatedData;
+
+    return Promise.resolve(updatedData);
+  } catch (e) {
+    console.error(`Could not GET UserActivity for session token ${ sessionTokenName }`, e); // eslint-disable-line no-console
+
+    return Promise.reject(e);
+  }
+}
+
+export async function updateUserActivityToken(store: any, sessionTokenName: string):Promise<userActivityResponse> {
+  const updateUserActivity = await store.dispatch('management/create', {
+    apiVersion: 'ext.cattle.io/v1',
+    kind:       'UserActivity',
+    type:       EXT.USERACTIVITY,
+    metadata:   { name: sessionTokenName },
+    spec:       { tokenId: sessionTokenName }
+  });
+
+  try {
+    const savedData = await updateUserActivity.save();
+
+    // update stored data
+    storedUserActivityResponse = savedData;
+
+    return Promise.resolve(savedData);
+  } catch (e) {
+    console.error(`Could not update (POST) UserActivity for session token ${ sessionTokenName }`, e); // eslint-disable-line no-console
+
+    return Promise.reject(e);
+  }
+}
+
+export function parseTTLData(userActivityData: userActivityResponse): parsedInactivitySetting {
+  const currDate = Date.now();
+  const endDate = new Date(userActivityData.status?.expiresAt).getTime();
+
+  // let's give this a 3 second buffer so that we can make sure the logout happens by the frontend
+  const thresholdSeconds = Math.floor((endDate - currDate) / 1000) - 3;
+
+  // Amount of time the user sees the inactivity warning
+  const courtesyTimerVal = Math.floor(thresholdSeconds * 0.1);
```

Comment:
I don't understand the 0.1 here. It looks like weren't evaluating everything in seconds already.

Reply:
Previous logic, which I reused. It was considered as "display time" of the modal 10% of the total time (with a max of modal display time of 5 min).

If you put a timeout of 1 minute, as general math it's expected for the timer to show modal to be 54sec and the modal will be displayed for 6sec.

This is open for debate as it's easily changed if you guys would like me to tweak it, but for now I'll just keep it as is and resolve this. Just unresolve and comment for an action point for me 🙏